homeassistant/components/alarmdecoder/sensor.py

```diff
@@ -6,25 +6,33 @@
 from homeassistant.helpers.dispatcher import async_dispatcher_connect
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
-from .const import SIGNAL_PANEL_MESSAGE
+from .const import DATA_AD, DOMAIN, SIGNAL_PANEL_MESSAGE
+from .entity import AlarmDecoderEntity
 
 
 async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up for AlarmDecoder sensor."""
 
-    entity = AlarmDecoderSensor()
+    client = hass.data[DOMAIN][entry.entry_id][DATA_AD]
+    entity = AlarmDecoderSensor(client=client)
     async_add_entities([entity])
 
 
-class AlarmDecoderSensor(SensorEntity):
+class AlarmDecoderSensor(AlarmDecoderEntity, SensorEntity):
     """Representation of an AlarmDecoder keypad."""
 
+    _attr_has_entity_name = True
```

Comment:
Not needed I think, as it comes from the base entity.
```suggestion
```

Reply:
You're correct.  I verified that it still works with this change